### PR TITLE
403 - add resource/package tracing query to helper functions

### PR DESCRIPTION
**Pull request by garrettH3S**

### CODE REVIEW

File: ckanext/bcgov/util/util.py [hunk 1]
```diff
@@ -475,3 +475,9 @@ def can_view_resource(resource):
                 return True
 
     return False
+
+def get_package_tracking(package):
+    return (model.TrackingSummary.get_for_package(package['id']))
```
Comment (jrods) [inline ckanext/bcgov/util/util.py]: Pass in just the `id` var rather then the dataset dict.

File: ckanext/bcgov/util/util.py [hunk 2]
```diff
@@ -475,3 +475,9 @@ def can_view_resource(resource):
...
+
+def get_resource_tracking(resource):
+    return (model.TrackingSummary.get_for_resource(resource['url']))
```
Comment (jrods) [inline ckanext/bcgov/util/util.py]: Same here, just pass in `url` from the resource dict

**Comment by garrettH3S** [comment]
This pr needs some server script run. 

`paster --plugin=ckan db upgrade -c <.ini file path>` in the startup`
needs to be added before the db-indexing
`paster --plugin=ckan search-index rebuild -c <.ini file path>`

also, a cron job needs to be registered to run that command every so often.  See this setup [page](http://docs.ckan.org/en/latest/maintaining/tracking.html)

File: ckanext/bcgov/util/util.py [hunk 3]
```diff
@@ -477,27 +478,46 @@ def can_view_resource(resource):
     return False
 
 def get_package_tracking(package_id):
-    # dl_resource_id = 'e0d0e432-0f86-4db9-8ea7-5a4e002c5c0f'
-    dl_resource_id = '1a7df2c1-f7df-46f1-a920-da85c04eeab6'
+
```
Comment (jrods) [inline ckanext/bcgov/util/util.py]: Remove commented block, lets try to keep the code base free of blocks of comments.

File: ckanext/bcgov/util/util.py [hunk 4]
```diff
@@ -26,6 +26,7 @@
 site_url = config.get('ckan.site_url')
 api_key = config.get('ckan.api_key')
 log = logging.getLogger('ckanext.bcgov.util')
+download_resource_id = config.get('ckan.datasource.downloads.resource')
```
Comment (jrods) [inline ckanext/bcgov/util/util.py]: Move this line into the get_package_tracking function

File: ckanext/bcgov/util/util.py (hunk 3 above)
Comment (garrettH3S) [reply ckanext/bcgov/util/util.py]: Oops, that was supposed to be for testing only.

File: ckanext/bcgov/util/util.py (hunk 4 above)
Comment (garrettH3S) [reply ckanext/bcgov/util/util.py]: done.

File: ckanext/bcgov/util/util.py (hunk 2 above)
Comment (garrettH3S) [reply ckanext/bcgov/util/util.py]: done

File: ckanext/bcgov/util/util.py (hunk 1 above)
Comment (garrettH3S) [reply ckanext/bcgov/util/util.py]: done.

File: ckanext/bcgov/plugin.py [hunk 5]
```diff
@@ -50,11 +54,13 @@
...
                                              remove_user_link,
                                              get_ofi_config,
-                                             get_ofi_resources
+                                             get_ofi_resources,
+                                             log_this
                                              )
 
 abort = base.abort
 
+log = logging.getLogger('ckan.ckanbcgov.plugin')
```
Comment (jrods) [inline ckanext/bcgov/plugin.py]: use `'ckanext.bcgov'` instead for the logger namespace

Comment (garrettH3S) [reply ckanext/bcgov/plugin.py]: updated.

**Pull request merged**